src/aleph/http/client_middleware.clj

```diff
@@ -920,9 +920,17 @@
       (opt req :save-request)
       (assoc :aleph/request req'))))
 
+(defn ^:no-doc wrap-validation [req]
+  (when-not (:request-method req)
+    (throw (IllegalArgumentException. ":request-method must be specified")))
+  (when-not (:request-url req)
+    (throw (IllegalArgumentException. ":request-url must be specified")))
```

Comment:
Should we consider malli or spec for this?

Also, at the top, I believe the `:method` key is also acceptable, but we have some middleware that converts it into `:request-method` later. That might need to be handled, don't recall where in the chain this falls.

Reply:
Yes, this is performed inside `wrap-url` and `wrap-method` which is performed just before `wrap-validation`.
I adapted to use spec instead.

Reply:
No experience with malli at all, I'm using spec as a habit. 
But I'm not even sure it's worth using one or the other inside Aleph actually...

Reply:
I'm fine with using them to validate user input. 

Also, there's apparently already a ring ns for validating ring maps: https://github.com/ring-clojure/ring/blob/8af4ab93190dfe5b4827c14b416a4cb92e18cdaf/ring-devel/src/ring/middleware/lint.clj, if we want to use it.

Reply:
I tried to create a spec from this `lint.clj` namespace, but I end up being a little bit more open to _exotic_ inputs.